models/api_host.go

```diff
@@ -1,27 +1,114 @@
 package models
 
-// APIHost - the host struct for API usage
-type APIHost struct {
+import "net"
+
+// ApiHost - the host struct for API usage
+type ApiHost struct {
 	ID              string   `json:"id"`
 	Verbosity       int      `json:"verbosity"`
 	FirewallInUse   string   `json:"firewallinuse"`
 	Version         string   `json:"version"`
-	IPForwarding    bool     `json:"ipforwarding"`
-	DaemonInstalled bool     `json:"daemoninstalled"`
-	HostPass        string   `json:"hostpass"`
 	Name            string   `json:"name"`
 	OS              string   `json:"os"`
-	Interface       string   `json:"interface"`
 	Debug           bool     `json:"debug"`
+	IsStatic        bool     `json:"isstatic"`
 	ListenPort      int      `json:"listenport"`
 	LocalAddress    string   `json:"localaddress"`
 	LocalRange      string   `json:"localrange"`
 	LocalListenPort int      `json:"locallistenport"`
 	ProxyListenPort int      `json:"proxy_listen_port"`
 	MTU             int      `json:"mtu" yaml:"mtu"`
 	Interfaces      []Iface  `json:"interfaces" yaml:"interfaces"`
+	EndpointIP      string   `json:"endpointip" yaml:"endpointip"`
 	PublicKey       string   `json:"publickey"`
 	MacAddress      string   `json:"macaddress"`
 	InternetGateway string   `json:"internetgateway"`
 	Nodes           []string `json:"nodes"`
+	ProxyEnabled    bool     `json:"proxy_enabled" yaml:"proxy_enabled"`
+}
+
+// Host.ConvertNMHostToAPI - converts a Netmaker host to an API editable host
+func (h *Host) ConvertNMHostToAPI() *ApiHost {
+	a := ApiHost{}
+	a.Debug = h.Debug
+	a.EndpointIP = h.EndpointIP.String()
+	a.FirewallInUse = h.FirewallInUse
+	a.ID = h.ID.String()
+	a.Interfaces = h.Interfaces
+	a.InternetGateway = h.InternetGateway.String()
+	if isEmptyAddr(a.InternetGateway) {
+		a.InternetGateway = ""
+	}
+	a.IsStatic = h.IsStatic
+	a.ListenPort = h.ListenPort
+	a.LocalAddress = h.LocalAddress.String()
+	if isEmptyAddr(a.LocalAddress) {
+		a.LocalAddress = ""
+	}
+	a.LocalListenPort = h.LocalListenPort
+	a.LocalRange = h.LocalRange.String()
+	if isEmptyAddr(a.LocalRange) {
+		a.LocalRange = ""
+	}
+	a.MTU = h.MTU
+	a.MacAddress = h.MacAddress.String()
+	a.Name = h.Name
+	a.OS = h.OS
+	a.Nodes = h.Nodes
+	a.ProxyEnabled = h.ProxyEnabled
+	a.ProxyListenPort = h.ProxyListenPort
+	a.PublicKey = h.PublicKey.String()
+	a.Verbosity = h.Verbosity
+	a.Version = h.Version
+
+	return &a
+}
+
+// APIHost.ConvertAPIHostToNMHost - convert's a given apihost struct to
+// a Host struct
+func (a *ApiHost) ConvertAPIHostToNMHost(currentHost *Host) *Host {
```

Comment:
why does this need a parameter?

Reply:
there are fields which are not editable, but we don't send to the UI, so we need to ensure they stay on the server